src/stats/rpc_stats.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "rpc/server.h"
+#include "stats/stats.h"
+#include "util.h"
+#include "utilstrencodings.h"
+
+#include <stdint.h>
+
+#include <univalue.h>
+
+UniValue getmempoolstats(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 0)
+        throw std::runtime_error(
+            "getmempoolstats\n"
+            "\nReturns the collected mempool statistics.\n"
+            "\nThe samples are segregated in multiple precision groups.\n"
+            "\nSamples time interval is not guaranteed to be constant, hence there\n"
+            "is a time delta in each sample relative to the last sample.\n"
+            "\nResult:\n"
+            "  [\n"
+            "    {\n"
+            "      \"sample_interval\"    : \"interval\",      (numeric) Interval target in seconds\n"
+            "      \"time_from\"          : \"timestamp\",     (numeric) Timestamp, first sample\n"
+            "      \"samples\"            : [\n"
+            "                                 [<delta_in_secs>,<tx_count>,<dynamic_mem_usage>,<min_fee_per_k>],\n"
+            "                                 [<delta_in_secs>,<tx_count>,<dynamic_mem_usage>,<min_fee_per_k>],\n"
+            "                                 ...\n"
+            "                             ]\n"
+            "    }\n"
+            "    ,...\n"
+            "  ]\n"
+            "\nExamples:\n" +
+            HelpExampleCli("getmempoolstats", "") + HelpExampleRpc("getmempoolstats", ""));
+
+    // get stats from the core stats model
+    uint64_t timeFrom = 0;
+    std::vector<unsigned int> groups = CStats::DefaultStats()->mempoolCollector->getPrecisionGroupsAndIntervals();
+
+    UniValue groupsUni(UniValue::VARR);
+    for (unsigned int i = 0; i < groups.size(); i++) {
+
+        MempoolSamplesVector samples = CStats::DefaultStats()->mempoolCollector->getSamplesForPrecision(i, timeFrom);
+
+        // use "flat" json encoding for performance reasons
+        UniValue samplesObj(UniValue::VARR);
+        for (const struct CStatsMempoolSample& sample : samples) {
+            UniValue singleSample(UniValue::VARR);
```

Comment:
In commit "Add mempool statistics collector"

It seems like it would be more user friendly and extensible if this were an object instead of an array, so the elements could be named. It's a little awkard to have to remember that the first element in the array is time delta, second element is transaction count, etc.

Reply:
I tend to agree. I would also use the timestamp directly instead of an offset.